server/src/main/java/org/opensearch/common/cache/stats/DefaultCacheStatsHolder.java

```diff
@@ -208,15 +210,14 @@ private ImmutableCacheStats removeDimensionsHelper(List<String> dimensionValues,
         return statsToDecrement;
     }
 
-    // pkg-private for testing
-    Node getStatsRoot() {
+    public Node getStatsRoot() {
         return statsRoot;
     }
 
     /**
      * Nodes that make up the tree in the stats holder.
      */
-    protected static class Node {
+    public static class Node {
         private final String dimensionValue;
         // Map from dimensionValue to the DimensionNode for that dimension value.
         final Map<String, Node> children;
```

Comment:
nit: Should this be private now since the getter is public? I am assuming we don't directly initialize children outside of the constructor?

Reply:
It can't be private since ImmutableCacheStatsHolder uses it during construction. But I can make it protected again. After reworking the TSC stats holder tests, it doesn't need to be public.